service/timestreamwrite/api_client.go

```diff
@@ -241,6 +261,84 @@ func addRetryMiddlewares(stack *middleware.Stack, o Options) error {
 	return retry.AddRetryMiddlewares(stack, mo)
 }
 
+// resolves EnableEndpointDiscovery configuration
+func resolveEnableEndpointDiscoveryFromConfigSources(cfg aws.Config, o *Options) error {
+	if len(cfg.ConfigSources) == 0 {
+		return nil
+	}
+	value, found, err := internalConfig.ResolveEnableEndpointDiscovery(context.Background(), cfg.ConfigSources)
+	if err != nil {
+		return err
+	}
+	if found {
+		o.EnableEndpointDiscovery = value
+	}
+	return nil
+}
+
+// resolves endpoint cache on client
+func resolveEndpointCache(c *Client) {
+	c.endpointCache = internalEndpointDiscovery.NewEndpointCache(internalEndpointDiscovery.DefaultEndpointCacheSize)
```

Comment:
To my earlier comment, I think it might make sense to just generate the default value inline into this call. Not sure if it makes sense for the endpointdiscovery package to export a constant for a default that it doesn't directly reference itself. Thoughts?

Reply:
I see - So if we want to change the default endpoint cache size we won't need to bump the endpoint-discovery module. 

My reasoning was to keep it as Go constant, instead of constant in java codegen.  Agree with your idea to avoid bumping endpoint-discovery module.